AppCenterDistribute/AppCenterDistributeTests/MSDistributeCheckForUpdateTests.m

```diff
@@ -3,112 +3,283 @@
 
 #import "MSBasicMachOParser.h"
 #import "MSChannelGroupProtocol.h"
+#import "MSDistributeIngestion.h"
 #import "MSDistributePrivate.h"
+#import "MSMockKeychainUtil.h"
+#import "MSMockUserDefaults.h"
 #import "MSTestFrameworks.h"
 #import "MSUtility+StringFormatting.h"
 
 static NSString *const kMSTestAppSecret = @"IAMSECRET";
 
 @interface MSDistributeCheckForUpdateTests : XCTestCase
 
+@property(nonatomic) MSDistribute *sut;
+@property(nonatomic) id settingsMock;
+@property(nonatomic) id keychainUtilMock;
+@property(nonatomic) id bundleMock;
+@property(nonatomic) id parserMock;
+@property(nonatomic) id utilityMock;
+
 @end
 
 @implementation MSDistributeCheckForUpdateTests
 
 - (void)setUp {
+  [super setUp];
+  self.settingsMock = [MSMockUserDefaults new];
+  self.keychainUtilMock = [MSMockKeychainUtil new];
+
+  // Utility mock.
+  self.utilityMock = OCMClassMock([MSUtility class]);
+  OCMStub(ClassMethod([self.utilityMock sha256:OCMOCK_ANY])).andReturn(@"RELEASEHASH");
+
+  // Bundle mock.
+  self.bundleMock = OCMClassMock([NSBundle class]);
+  OCMStub([self.bundleMock mainBundle]).andReturn(self.bundleMock);
+
+  // Parser mock.
+  id parserMock = OCMClassMock([MSBasicMachOParser class]);
+  OCMStub([parserMock machOParserForMainBundle]).andReturn(parserMock);
+  OCMStub([parserMock uuid]).andReturn([[NSUUID alloc] initWithUUIDString:@"CD55E7A9-7AD1-4CA6-B722-3D133F487DA9"]);
+
+  // Distribute instance.
   [MSDistribute resetSharedInstance];
+  self.sut = [MSDistribute sharedInstance];
+}
+
+- (void)tearDown {
+  [super tearDown];
+  [self.settingsMock stopMocking];
+  [self.keychainUtilMock stopMocking];
+  [self.bundleMock stopMocking];
+  [self.parserMock stopMocking];
+  [self.utilityMock stopMocking];
+  self.sut = nil;
```

Comment:
Add `resetSharedInstance` to `tearDown`

Reply:
We shouldn't rely on clean up that's done by other tests. We should do it in `setUp` wherever it is required.